### ISSUE TITLE
Create a CLI package to make it easier to install React Navigation packages

### ISSUE THREAD

**Issue by satya164** [react-navigation/react-navigation#8384]
Currently we need to install a lot of different packages, so installation ca get long. A CLI would allow us to streamline the experience.

Example usage:

```sh
# Install stack navigator
npx @react-navigation/cli add stack
```

This should install the navigator along with all of the required peer dependencies.

**Comment by Aung-Myint-Thein** [comment]
Would also be nice if we can have a table of packages and their latest version numbers. I am trying to update from 5.0.7 but struggling to find out what is the latest version numbers. :D

**Comment by saideepesh000** [comment]
@satya164, hey I would do a prototype kinda thing and send it to u.

**Comment by saideepesh000** [comment]
@satya164, @brentvatne, @skevy  I kinda created what you mentioned, but just made to decide what they want to install, it isn't published to npm or uploaded to GitHub, if you wish I want to continue on this, I would do that. Thanks.

![Screen Recording (28-10-2020 18-19-52)](https://user-images.githubusercontent.com/43727167/97439785-a4eaeb00-194c-11eb-906f-f36eb983602a.gif)

**Comment by satya164** [comment]
@saideepesh000 looks nice. I don't have much time right now, but can you open a PR so I can take a look when possible? You can create a new folder called `cli` under the `packages` folder.

For initial version I'd like to have what I mentioned so we can mention it as an one-liner in the documentation. The command should do the following:

- Based on the argument (e.g. `stack`), it would check for a package named `@react-navigation/stack` on npm, and fetch it's metadata, or display error if the package doesn't exist
- It can read the peer dependencies from the metadata and install all packages with one command
- It should use choose which command to use to install based on following:
	- Use `expo` if we find `expo` in the `dependencies` of `package.json`
	- Use `yarn` if the user has `yarn.lock` file
	- Use `npm` if the user has `package-json.lock`
    - If no lockfile is present, use `yarn` if it's installed, otheriwse `npm`
- For gesture handler, we need to add `import 'react-native-gesture-handler'`. Right now it's the only package which needs it, so we can hardcode it to check if it was installed, and print a message about adding it, or even add it the top of `index.js`/`index.ts` automatically. 

Selecting packages to install from a liist and updating what's installed can be secondary functionality. Let's discuss them after the basic functionality. Also, we'd likely want to not display packages such as `core` or `routers` since normally users don't need to install them and they can confuse people.

cc @brentvatne if you have some ideas, please comment

**Comment by saideepesh000** [comment]
Thanks, @satya164, I would do as my as I can from whatever you mentioned above and raise a PR.

**Comment by saideepesh000** [comment]
@satya164, So far I completed following;
1. Fetching the peerDependencies data through (https://registry.npmjs.org/@react-navigation/${package}/latest) and installing those.
2. Finding for dependency "expo", finding package-lock or yarn lock.

To Do:
1. Making it modular, readable(so far I wrote everything in a single file) and changing to Typescript.
2. Checking if dependencies are already installed.
3. Importing "react-native-gesture-handler" automatically.

Soon I'll PR after making it modular, readable, and Typed(😂).
